### PR TITLE
Add icons on buttons in logout and shutdown dialog

### PR DESCRIPTION
**Pull request by tamplan**
Change cancel icon to 'gtk-cancel'
Add some icons to others buttons.

After PR, the dialogs are like this (sorry in French):
Logout dialog
Capture du 2020-06-22 13-46-52
Shutdown dialog
Capture du 2020-06-22 13-45-30

Regards
[Edit]Icons are changed but screenshot not.[/Edit]

### CODE REVIEW

File: mate-session/gsm-logout-dialog.c [hunk 1]
```diff
@@ -443,17 +443,17 @@ gsm_get_dialog (GsmDialogLogoutType type,
...
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("_Switch User"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("_Switch User"), "system-users",
                                                GSM_LOGOUT_RESPONSE_SWITCH_USER);
                 }
 
                 gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                            _("_Cancel"), "process-stop",
+                                            _("_Cancel"), "gtk-cancel",
```
Comment (sc0w) [inline mate-session/gsm-logout-dialog.c:452]: `gtk-cancel` is deprecated, `process-stop` is fine

File: mate-session/gsm-logout-dialog.c [hunk 2]
```diff
@@ -464,30 +464,30 @@ gsm_get_dialog (GsmDialogLogoutType type,
...
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("_Restart"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("_Restart"), "gtk-refresh",
                                                GSM_LOGOUT_RESPONSE_REBOOT);
                 }
 
                 gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                            _("_Cancel"), "process-stop",
+                                            _("_Cancel"), "gtk-cancel",
```
Comment (sc0w) [inline mate-session/gsm-logout-dialog.c:485]: `gtk-cancel` is deprecated

File: mate-session/gsm-logout-dialog.c [hunk 3]
```diff
@@ -464,30 +464,30 @@ gsm_get_dialog (GsmDialogLogoutType type,
...
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("_Hibernate"), "system-hibernate",
                                                GSM_LOGOUT_RESPONSE_HIBERNATE);
                 }
 
                 if (gsm_logout_supports_reboot (logout_dialog)) {
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("_Restart"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("_Restart"), "gtk-refresh",
```
Comment (sc0w) [inline mate-session/gsm-logout-dialog.c:480]: `gtk-refresh` is deprecated, you can use `view-refresh` instead, all the icon names with `"gtk-..."` in the name are deprecated

File: mate-session/gsm-logout-dialog.c [hunk 4]
```diff
@@ -464,30 +464,30 @@ gsm_get_dialog (GsmDialogLogoutType type,
                 logout_dialog->default_response = GSM_LOGOUT_RESPONSE_SHUTDOWN;
 
                 if (gsm_logout_supports_system_suspend (logout_dialog)) {
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("S_uspend"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("S_uspend"), "system-suspend",
```
Comment (sc0w) [inline mate-session/gsm-logout-dialog.c:468]: in my tests, this icon is missing

File: mate-session/gsm-logout-dialog.c [hunk 5]
```diff
@@ -464,30 +464,30 @@ gsm_get_dialog (GsmDialogLogoutType type,
...
                                                GSM_LOGOUT_RESPONSE_SLEEP);
                 }
 
                 if (gsm_logout_supports_system_hibernate (logout_dialog)) {
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("_Hibernate"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("_Hibernate"), "system-hibernate",
```
Comment (sc0w) [inline mate-session/gsm-logout-dialog.c:474]: in my tests, this icon is missing

File: mate-session/gsm-logout-dialog.c (hunk 1 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c:452]: The button label is cancel, i have seen a Mate application with the "correct" icon (red circle with white cross on it).
Sure its name is not "gtk-cancel" but another one. Updated soon.
Thanks.

File: mate-session/gsm-logout-dialog.c (hunk 2 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c:485]: Same as above here, thanks again.

File: mate-session/gsm-logout-dialog.c (hunk 3 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c:480]: Ok for "view-refresh", updated soon, thanks.

File: mate-session/gsm-logout-dialog.c (hunk 4 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c:468]: I'll try with another set of icons like "Mate" or "Menta" to find another one.  I search for a icon with name that not begins with "gnome", perhaps a good idea for the future...

File: mate-session/gsm-logout-dialog.c (hunk 5 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c:474]: Same as above, updated soon, thanks for all again.

File: mate-session/gsm-logout-dialog.c (hunk 4 above)
Comment (sc0w) [reply mate-session/gsm-logout-dialog.c:468]: In debian we have the package `gtk-3-examples` , it installs the  program `gtk3-icon-browser` and you can see a lot of icons

for reference:
https://askubuntu.com/questions/695796/view-list-of-all-available-unique-icons-with-their-names-and-thumbnail

@mate-desktop/core-team the icon names `"system-suspend"` and `"system-hibernate"` work for you? not here

File: mate-session/gsm-logout-dialog.c (hunk 1 above)
Comment (sc0w) [reply mate-session/gsm-logout-dialog.c:452]: `gtk-cancel` and `process-stop` are the same icon, but `gtk-cancel` is deprecated, so... is better `process-stop`

Comment (tamplan) [reply mate-session/gsm-logout-dialog.c:452]: I have different icons between process-stop and the icon used in Caja for example:
Capture du 2020-06-24 07-31-40
mate-screenshot uses the same than Caja, this is the icon i purpose to use.

Comment (tamplan) [reply mate-session/gsm-logout-dialog.c:452]: I have these icons:
Capture du 2020-06-24 07-39-52
Without the PR applied, the icon used in the shutdown dialog is one of the first line, all others applications use one of the second. I have read the correct name in the original code. I must investigate if the icon theme is correctly loaded in some applications and not here.

Did you have the same issue, icons different between Caja or Mate-screenshot and here?

Thanks.

Comment (tamplan) [reply mate-session/gsm-logout-dialog.c:452]: Using tk3-icon-browser, the icon "process-stop" is the good one, one of the second line.

Comment (sc0w) [reply mate-session/gsm-logout-dialog.c:452]: caja and mate-screenshot have the same icon here, process-stop (the same as gtk-cancel):



the icon with X seems to be the icon name `"edit-delete"`

just keep the lines `process-stop` with no changes

Comment (tamplan) [reply mate-session/gsm-logout-dialog.c:452]: Could you give me the name of the icon set you use, please?
Thanks.

Comment (sc0w) [reply mate-session/gsm-logout-dialog.c:452]: mate-icon-theme

**Comment by tamplan** [comment]
I don't understand where i'm wrong, i have just another commit https://github.com/tamplan/mate-session-manager/commit/c96548100730391603b56aca5900e8fbdaf5e9b3
The PR here is not updated...

File: mate-session/gsm-logout-dialog.c (hunk 5 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Done, switch to another one.

File: mate-session/gsm-logout-dialog.c (hunk 4 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Thanks for the link!
Done, switch to another one.

File: mate-session/gsm-logout-dialog.c (hunk 3 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Done, switch to "view-refresh".

File: mate-session/gsm-logout-dialog.c (hunk 2 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Done, switch back to "process-stop".

File: mate-session/gsm-logout-dialog.c (hunk 1 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Lol, I find where the bug was: my brain! I was perhaps switch to another icon set for testing and i was sure the "process-stop" was different, sorry for the waste of time...
Done, switch back to "process-stop".

File: mate-session/gsm-logout-dialog.c [hunk 6]
```diff
@@ -443,17 +443,17 @@ gsm_get_dialog (GsmDialogLogoutType type,
                 logout_dialog->default_response = GSM_LOGOUT_RESPONSE_LOGOUT;
 
                 if (gsm_logout_supports_switch_user (logout_dialog)) {
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("_Switch User"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("_Switch User"), "system-users",
```
Comment (raveit65) [inline mate-session/gsm-logout-dialog.c:447]: Indent!

File: mate-session/gsm-logout-dialog.c [hunk 7]
```diff
@@ -443,17 +443,17 @@ gsm_get_dialog (GsmDialogLogoutType type,
...
                 }
 
                 gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
                                             _("_Cancel"), "process-stop",
                                             GTK_RESPONSE_CANCEL);
 
-                gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                       _("_Log Out"),
+                gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                       _("_Log Out"), "system-log-out",
```
Comment (raveit65) [inline mate-session/gsm-logout-dialog.c:456]: Indent!

File: mate-session/gsm-logout-dialog.c [hunk 8]
```diff
@@ -464,20 +464,20 @@ gsm_get_dialog (GsmDialogLogoutType type,
                 logout_dialog->default_response = GSM_LOGOUT_RESPONSE_SHUTDOWN;
 
                 if (gsm_logout_supports_system_suspend (logout_dialog)) {
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("S_uspend"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("S_uspend"), "battery",
```
Comment (raveit65) [inline mate-session/gsm-logout-dialog.c:468]: Indent!

File: mate-session/gsm-logout-dialog.c [hunk 9]
```diff
@@ -464,20 +464,20 @@ gsm_get_dialog (GsmDialogLogoutType type,
...
                                                GSM_LOGOUT_RESPONSE_SLEEP);
                 }
 
                 if (gsm_logout_supports_system_hibernate (logout_dialog)) {
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("_Hibernate"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("_Hibernate"), "drive-harddisk",
```
Comment (raveit65) [inline mate-session/gsm-logout-dialog.c:474]: Indent!

File: mate-session/gsm-logout-dialog.c [hunk 10]
```diff
@@ -464,20 +464,20 @@ gsm_get_dialog (GsmDialogLogoutType type,
...
                                                GSM_LOGOUT_RESPONSE_HIBERNATE);
                 }
 
                 if (gsm_logout_supports_reboot (logout_dialog)) {
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("_Restart"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("_Restart"), "view-refresh",
```
Comment (raveit65) [inline mate-session/gsm-logout-dialog.c:480]: Indent!

File: mate-session/gsm-logout-dialog.c [hunk 11]
```diff
@@ -486,8 +486,8 @@ gsm_get_dialog (GsmDialogLogoutType type,
                                             GTK_RESPONSE_CANCEL);
 
                 if (gsm_logout_supports_shutdown (logout_dialog)) {
-                        gtk_dialog_add_button (GTK_DIALOG (logout_dialog),
-                                               _("_Shut Down"),
+                        gsm_util_dialog_add_button (GTK_DIALOG (logout_dialog),
+                                               _("_Shut Down"), "system-shutdown",
```
Comment (raveit65) [inline mate-session/gsm-logout-dialog.c:490]: Indent!

File: mate-session/gsm-logout-dialog.c (hunk 7 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Done!

File: mate-session/gsm-logout-dialog.c (hunk 8 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Done!

File: mate-session/gsm-logout-dialog.c (hunk 9 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Done!

File: mate-session/gsm-logout-dialog.c (hunk 10 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Done!

File: mate-session/gsm-logout-dialog.c (hunk 11 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Done!

File: mate-session/gsm-logout-dialog.c (hunk 6 above)
Comment (tamplan) [reply mate-session/gsm-logout-dialog.c]: Done!

**Comment by raveit65** [comment]
```
[rave@mother mate-session-manager]$ git branch 
  1.20
  1.22
  1.24
* master
  tamplan-icons
[rave@mother mate-session-manager]$ git checkout tamplan-icons 
Switched to branch 'tamplan-icons'
[rave@mother mate-session-manager]$ git log
commit d558c6095c85f274243c911e8dea8516e7d7f06f (HEAD -> tamplan-icons)
Author: tamplan <tamplan@free.fr>
Date:   Sat Jun 27 19:07:03 2020 +0200

    Change some icons by other always available

commit 95f60f053f10034f58b26b1f7db4723ad9359ef8
Author: tamplan <tamplan@free.fr>
Date:   Wed Jun 24 20:28:56 2020 +0200

    Change some icons by other always available

commit 26ced0be364ed3be106cfe0081502123ce2759fc (origin/master, origin/HEAD, master)
Author: rbuj <robert.buj@gmail.com>
Date:   Sun Jun 21 12:30:27 2020 +0200

    Using #include with system libraries
```
2 commits?
https://patch-diff.githubusercontent.com/raw/mate-desktop/mate-session-manager/pull/253.patch
Why we should accept this?

**Comment by tamplan** [comment]
> 2 commits?
> https://patch-diff.githubusercontent.com/raw/mate-desktop/mate-session-manager/pull/253.patch
> Why we should accept this?

Sorry, forgot to squash and rebase.

**Comment by tamplan** [comment]
> Works for me.

Thanks you very much!

**Comment by tamplan** [comment]
> ```
> [rave@mother mate-session-manager]$ git branch 
>   1.20
>   1.22
>   1.24
> * master
>   tamplan-icons
> [rave@mother mate-session-manager]$ git checkout tamplan-icons 
> Switched to branch 'tamplan-icons'
> [rave@mother mate-session-manager]$ git log
> commit d558c6095c85f274243c911e8dea8516e7d7f06f (HEAD -> tamplan-icons)
> Author: tamplan <tamplan@free.fr>
> Date:   Sat Jun 27 19:07:03 2020 +0200
> 
>     Change some icons by other always available
> 
> commit 95f60f053f10034f58b26b1f7db4723ad9359ef8
> Author: tamplan <tamplan@free.fr>
> Date:   Wed Jun 24 20:28:56 2020 +0200
> 
>     Change some icons by other always available
> 
> commit 26ced0be364ed3be106cfe0081502123ce2759fc (origin/master, origin/HEAD, master)
> Author: rbuj <robert.buj@gmail.com>
> Date:   Sun Jun 21 12:30:27 2020 +0200
> 
>     Using #include with system libraries
> ```
> 
> 2 commits?
> https://patch-diff.githubusercontent.com/raw/mate-desktop/mate-session-manager/pull/253.patch
> Why we should accept this?

Hi,
I have this:
```Branche : icons
$ git log
commit 18825d3cd870e418cc894e6e0306029e1a1f4900 (HEAD -> icons, origin/icons)
Author: tamplan <tamplan@free.fr>
Date:   Wed Jun 24 20:28:56 2020 +0200

    Change some icons by other always available

commit 454dccb8138d611da9798bde106363fa9cf83e01 (upstream/master, origin/master, origin/HEAD)
Author: tamplan <tamplan@free.fr>
Date:   Mon May 18 18:59:12 2020 +0200

    Add tab scrolling support for GTK3

commit 4a62ca15be595c521f3313c20dd9d4c415970d6e
Author: Pablo Barciela <scow@riseup.net>
Date:   Sun Apr 26 04:41:59 2020 +0200

    Travis CI: deploy: key github-token is an alias for token, using token
```

**Comment by raveit65** [comment]
> Sorry, forgot to squash and rebase.

> Hi,
> I have this:

???
Did you forgot that you fixed the double commits 4 days ago for yourself?

**Comment by tamplan** [comment]
> > Sorry, forgot to squash and rebase.
> 
> > Hi,
> > I have this:
> 
> ???
> Did you forgot that you fixed the double commits 4 days ago for yourself?

Sorry, you are right.
I would like to indicate that the PR no need changes actually and i can't see the resolve conversation button here.

**Comment by raveit65** [comment]
You will never see on a normal posting such a button, only with review comment.
But i approved it, which means my concern is solved.
Now we waiting for another reviewer :)

PS: no , need to answer, to avoid unneeded notifications in everybody's mailbox ;)

**Pull request merged**